### ISSUE TITLE
Lua: txn.http.req_rep_value doesn't work anymore

### ISSUE THREAD

**Issue by surki** [haproxy/haproxy#124]
```
HA-Proxy version 2.0-dev7 2019/06/11 - https://haproxy.org/
```

## What's the configuration?

haproxy.cfg:
```
...
...
use_backend %[lua.foo]
...
...
```

lua:
```
...
...
function foo(txn)
   ...
   ...
   txn.http:req_rep_value("Host", "(.*)-shadow$", "\\1")
   ...
   ...
end
core.register_fetches("foo", foo)
...
...
```

## Steps to reproduce the behavior

1.  Make a curl request with `curl -v -H 'Host: mydomain.com-shadow' http://localhost:80/`
2. Verify on the server that host is "mydomain.com" (i.e., `-shadow` removed)

## Actual behavior

Host header is not replaced (i.e., `mydomain.com-shadow` should become `mydomain.com`

## Expected behavior

Host header should be replaced (i.e., `mydomain.com-shadow` should become `mydomain.com`

## Do you have any idea what may have caused this?
If I place the same rule in haproxy.cfg, it seems to work.
It appears this rule in haproxy.cfg uses new HTX code flow, whereas lua goes through *legacy functions, which doesn't seem to work

## Do you have an idea how to solve the issue?
Probably make Lua also use HTX?

**Comment by wtarreau** [comment]
Hi,

thanks for your report. Lua also uses HTX, however the bindings between Lua and the rest of the code are much trickier and it is possible that some functions like this one do not work well anymore. We'll have a look, thank you!

**Comment by capflam** [comment]
The fix was backported to 2.0 and 1.9. It should work now. Thanks !

**Comment by surki** [comment]
Works now, thanks
